boot/freeldr/freeldr/lib/debug.c

```diff
@@ -203,10 +210,17 @@ VOID DebugPrintChar(UCHAR Character)
 
     if (DebugPort & RS232)
     {
+#ifndef _M_ARM
         if (Character == '\n')
             Rs232PortPutByte('\r');
 
         Rs232PortPutByte(Character);
+#else
+        if (Character == '\n')
+            ARMWriteToUART('\r');
```

Comment:
I wonder whether it could be of interest to alias `Rs232PortPutByte` to `ARMWriteToUART` and `Rs232PortInitialize` to a dummy function, in order to avoid those if(n)defs here -- at the expense of having them in the rs232.c file or, in the corresponding header? (that latter option may be slightly less invasive.)